sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,117 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import DVR
+
+public extension Filter {
+    
+    static let subscriptionIDRegex = try! NSRegularExpression(pattern: "(/(subscriptions))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: .caseInsensitive)
+    static let graphSubscriptionIDRegex = try! NSRegularExpression(pattern: "(https://(graph.windows.net))/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}", options: .caseInsensitive)
+    
+    static var subcriptionIDReplacement = "00000000-0000-0000-0000-000000000000"
+    
+    static var standardHeaderScrubbing: [String : FilterBehavior] = [
+        "location": .closure({ key, val in
+            scrubSubscriptionId(from: val ?? "")
+        }),
+        "operation-location": .remove,
+        "azure-asyncoperation": .remove,
+        "www-authenticate": .remove,
+        "access_token": .remove
+        
+    ]
+    
+    /// Returns the original string with subscription ID info (if any) replaced.
+    static func scrubSubscriptionId(from string: String) -> String {
+        var editedString = string
+        editedString.replaceAllMatches(usingRegex: Filter.subscriptionIDRegex, withReplacement: "$1/\(Filter.subcriptionIDReplacement)")
+        editedString.replaceAllMatches(usingRegex: Filter.graphSubscriptionIDRegex, withReplacement: "$1/\(Filter.subcriptionIDReplacement)")
+        
+        return editedString
+        
+        
+    }
+    
+    /// Attempts to scrub subscription ID from optional String. Returns nil if the input is nil.
+    static func tryScrubSubscriptionId(from string: String?) -> String? {
+        
+        if string != nil {
+            return scrubSubscriptionId(from: string!)
+        }
+        return nil
+    }
+    
+    /// Attempts to scrub subsctiption ID from optional Data that represents a string. Returns null if the input is nil or if the Data is not String-representable.
+    static func tryScrubSubscriptionId(from data: Data?) -> Data? {
+        
+        guard let stringData = String(data: data, encoding: .utf8) else {
+            return nil
+        }
+        return scrubSubscriptionId(from: stringData).data(using: .utf8)
+    }
```

Comment:
These methods preserve the same ambiguity that made the original implementation suspect. Just refactor `scrubScubscriptionIDs` to early out if it is unable to get to an actual usable string instead of chaining optionals.

Reply:
Maybe we should have a call because I'm not understanding the ambiguity you're talking about. If I don't allow the function to take an optional then all that means is everywhere I use it I'd have to guard against nil which is definitely an option if you'd prefer that route but it seems like that way adds more unnecessary code. I want to understand better the ambiguity you are mentioning.